code/datums/spells/alien_spells/corrosive_acid_spit.dm

```diff
@@ -0,0 +1,30 @@
+/obj/effect/proc_holder/spell/touch/alien_spell/corrosive_acid
+	name = "Corrosive acid"
+	desc = "Spit acid on someone in range, this acid melts through nearly anything and heavily damages anyone lacking proper safety equipment."
+	hand_path = "/obj/item/melee/touch_attack/alien/corrosive_acid"
+	action_icon_state = "alien_acid"
+	plasma_cost = 200
+
+/obj/item/melee/touch_attack/alien/corrosive_acid
+	name = "Corrosive acid"
+	desc = "A fistfull of death."
+	icon_state = "alien_acid"
+
+/obj/item/melee/touch_attack/alien/corrosive_acid/afterattack(atom/target, mob/living/carbon/user, proximity)
+	if(target == user)
+		to_chat(user, "<span class='noticealien'>You withdraw your readied acid.</span>")
+		..()
+		return
+	if(!proximity || isalien(target) || !iscarbon(user) || HAS_TRAIT(user, TRAIT_HANDS_BLOCKED)) // Don't want xenos ditching out of cuffs
+		return
+	plasma_check(200, user)
```

Comment:
Wouldn't it be better to just have this proc return a value? Instead of using `continue_cast`

Reply:
Done